tools/changelog-generator/app/generate-release-changelog.sh

```diff
@@ -9,10 +9,15 @@ fi
 
 source ${APP_PATH}/config-setup.sh
 
-REVISION=$(git rev-list --tags --max-count=1)
-LAST_VERSION_TAG=$(git describe --tags ${REVISION})
-echo "Getting new changes starting from the '${LAST_VERSION_TAG}' tag..."
 mkdir -p ${RELEASE_CHANGELOG_FILE_DIRECTORY}
-lerna-changelog --from=${LAST_VERSION_TAG} | sed -e "s/## Unreleased/## ${LATEST_VERSION}/g" > ${RELEASE_CHANGELOG_FILE_PATH}
+if [ -n "$FROM_TAG" ]; then
+    echo "Getting new changes starting from the '${FROM_TAG}' tag to '${LATEST_VERSION}' tag..."
+    lerna-changelog --from=${FROM_TAG} --to=${LATEST_VERSION} | sed -e "s/## Unreleased/## ${LATEST_VERSION}/g" > ${RELEASE_CHANGELOG_FILE_PATH}
+else
+    REVISION=$(git rev-list --tags --max-count=1)
```

Comment:
You can simplify this fragment greatly. Here's the pseudocode:

```
if ($FROM_TAG) {
TAG = FROM_TAG
} else {
REVISION=$(git rev-list --tags --max-count=1)
LAST_VERSION_TAG=$(git describe --tags ${REVISION});
TAG=LATEST_VERSION_TAG
}

echo "Getting new changes starting from the '${TAG}' tag..."
    lerna-changelog --from=${TAG} | sed -e "s/## Unreleased/## ${LATEST_VERSION}/g" > ${RELEASE_CHANGELOG_FILE_PATH}

```

Reply:
Another lines are also different. In first condition lerna-changelog is run with "from" and "to" arguments, in second condition only with "from". Echo messages are also different.